app/helpers/halalas.rb

```diff
@@ -0,0 +1,24 @@
+# coding: utf-8
+# frozen_string_literal: true
+class Halalas
+  CURRENCY= ' SAR'
+
+  def initialize value
+    @halalas= Integer(value)
+    @riyals= Rational(@halalas, 100)
+  end
+
+  def to_s
+    riyals_string + CURRENCY
+  end
+
+private
+
+  def riyals_string
```

Comment:
Consider blurring the line between data and behaviour. When data is accessed by message sending it insulates you from future changes.

```
attr_reader :riyals

def riyals_string
  if riyals < 0
    '(%.2f)' % riyals.abs
  else
     '%.2f' % riyals
   end
end
```